src/allmydata/util/jsonbytes.py

```diff
@@ -13,34 +13,41 @@
 if PY2:
     from future.builtins import filter, map, zip, ascii, chr, hex, input, next, oct, open, pow, round, super, bytes, dict, list, object, range, str, max, min  # noqa: F401
 
-
 import json
 
 
+def _bytes_to_unicode(obj):
+    """Convert any bytes objects to unicode, recursively."""
+    if isinstance(obj, bytes):
+        return obj.decode("utf-8")
+    if isinstance(obj, dict):
+        new_obj = {}
+        for k, v in obj.items():
+            if isinstance(k, bytes):
+                k = k.decode("utf-8")
```

Comment:
JSON only allows string keys, so that just errors out (though turns out Python's json support does an awful thing where it stringifies ints/floats/etc.. Not tuples or lists though).